locator/abstract_replication_strategy.cc

```diff
@@ -187,8 +200,9 @@ abstract_replication_strategy::get_ranges(inet_address ep, token_metadata_ptr tm
 
 dht::token_range_vector
 effective_replication_map::get_primary_ranges(inet_address ep) const {
-    return do_get_ranges([ep] (inet_address_vector_replica_set eps) {
-        return eps.size() > 0 && eps[0] == ep;
+    return do_get_ranges([ep] (bool& add_range, const inet_address& e) {
+        add_range = (e == ep);
+        return stop_iteration::yes;
```

Comment:
Again I'm really confused what this means - it seems like an error but maybe I'm just mistunderstanding.... Why stop after one iteration, don't we need to look at all the ranges?

Reply:
Previously, we also checked only the first entry:
```
        return eps.size() > 0 && eps[0] == ep;
```

The left part of the expression above is covered by the fact this functor is now called, meaning the vector has at least one entry,
The right part is covered by determining `add_value = (e == ep)` once, and stopping the iteration.